### PR TITLE
[charts] Improve default tooltip content 

### PR DESCRIPTION
**Pull request by oliviertassinari**
A few opportunities I saw looking randomly at the code.

### CODE REVIEW

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts [hunk 1]
```diff
@@ -27,10 +27,10 @@ export const ChartsTooltipRow = styled('tr', {
   name: 'MuiChartsTooltip',
   slot: 'Row',
 })(({ theme }) => ({
-  'tr:first-of-type& td': {
+  '& tr:first-of-type& td': {
```
Comment (LukasTy) [inline packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:30]: Shouldn't there be no space here to correctly target what we expect? 🤔 
```suggestion
  '&tr:first-of-type& td': {
```

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts [hunk 2]
```diff
@@ -27,10 +27,10 @@ export const ChartsTooltipRow = styled('tr', {
...
     paddingTop: theme.spacing(1),
   },
-  'tr:last-of-type& td': {
+  '& tr:last-of-type& td': {
```
Comment (LukasTy) [inline packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:33]:
```suggestion
  '&tr:last-of-type& td': {
```

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts [hunk 3]
```diff
@@ -41,19 +41,17 @@ export const ChartsTooltipCell = styled('td', {
...
   [`&.${chartsTooltipClasses.labelCell}`]: {
     paddingLeft: theme.spacing(1),
   },
   [`&.${chartsTooltipClasses.valueCell}`]: {
     paddingLeft: theme.spacing(4),
     color: (theme.vars || theme).palette.text.primary,
   },
-
-  'td:first-of-type&': {
+  '& td:first-of-type&': {
```
Comment (LukasTy) [inline packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:51]:
```suggestion
  '&td:first-of-type&': {
```

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts [hunk 4]
```diff
@@ -41,19 +41,17 @@ export const ChartsTooltipCell = styled('td', {
...
     paddingLeft: theme.spacing(2),
   },
-  'td:last-of-type&': {
+  '& td:last-of-type&': {
```
Comment (LukasTy) [inline packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:54]:
```suggestion
  '&td:last-of-type&': {
```

**Review by LukasTy — DISMISSED** [review]
The current change introduces a regression. 🙈

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts (hunk 1 above)
Comment (oliviertassinari) [reply packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:30]: Ah wait, I missed the &

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts (hunk 4 above)
Comment (oliviertassinari) [reply packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:54]: Oh my bad, I can't do this, I didn't notice the `&` in the end.

File: packages/x-charts/src/ChartsTooltip/DefaultChartsItemTooltipContent.tsx [hunk 5]
```diff
@@ -41,13 +41,11 @@ function DefaultChartsItemTooltipContent<T extends ChartSeriesType = ChartSeries
         <tbody>
           <ChartsTooltipRow className={classes.row}>
             <ChartsTooltipCell className={clsx(classes.markCell, classes.cell)}>
-              <ChartsTooltipMark ownerState={{ color }} className={classes.mark} />
+              <ChartsTooltipMark color={color} className={classes.mark} />
```
Comment (oliviertassinari) [inline packages/x-charts/src/ChartsTooltip/DefaultChartsItemTooltipContent.tsx:44]: This isn't `ownerState`

File: packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx [hunk 6]
```diff
@@ -50,16 +50,13 @@ function DefaultChartsAxisTooltipContent(props: ChartsAxisContentProps) {
                 <ChartsTooltipRow key={id} className={classes.row}>
                   <ChartsTooltipCell className={clsx(classes.markCell, classes.cell)}>
                     <ChartsTooltipMark
-                      ownerState={{ color: getColor(dataIndex) ?? color }}
-                      boxShadow={1}
```
Comment (oliviertassinari) [inline packages/x-charts/src/ChartsTooltip/DefaultChartsAxisTooltipContent.tsx:54]: It doesn't feel like the UI should be different based on the type of tooltip.

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
 export const ChartsTooltipPaper = styled('div', {
   name: 'MuiChartsTooltip',
   slot: 'Container',
 })(({ theme }) => ({
+  boxShadow: theme.shadows[1],
   backgroundColor: (theme.vars || theme).palette.background.paper,
   color: (theme.vars || theme).palette.text.primary,
   transition: theme.transitions.create('box-shadow'),
-  border: `1px solid ${(theme.vars || theme).palette.divider}`,
   borderRadius: theme.shape.borderRadius,
```
Comment (oliviertassinari) [inline packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:13]: Get it closer to how https://mui.com/material-ui/react-paper/ feels.

File: packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts [hunk 8]
```diff
@@ -58,15 +56,16 @@ export const ChartsTooltipCell = styled('td', {
   },
 }));
 
-// eslint-disable-next-line material-ui/no-styled-box
-export const ChartsTooltipMark = styled(Box, {
+export const ChartsTooltipMark = styled('div', {
```
Comment (oliviertassinari) [inline packages/x-charts/src/ChartsTooltip/ChartsTooltipTable.ts:59]: Avoid expensive work, div is much faster than Box.

**Review by alexfauquette — APPROVED** [review]

**Pull request merged**